Fixing bonnie test

Naresh-ibm opened the pull request:
Tool creates and deletes sequential and random files in a dir

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

Naresh-ibm commented:
The previous script was not running in all the systems so I have added a option parameter "data_size_to_pass" (-s with 0) to the command that script uses to run the bonnie.  the script was giving error of "no space left on disk" and fails. After adding the the option -s0 the script runs without any issue.
attached the log files with and without parameter for your reference.

[bonnie-fail-job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/663025/bonnie-fail-job.txt)
[pass-with-paramter-bonnie.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/663029/pass-with-paramter-bonnie.txt)

balamuruhans commented on perf/bonnie.py:
```diff
@@ -50,21 +56,25 @@ def setUp(self):
 
         cmd = ('%s/bonnie++ %s' % (self.srcdir, " ".join(args)))
-        process.run(cmd)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
```
`if process.system(cmd, shell=True, ignore_status=True):` will be sufficient

balamuruhans commented on perf/bonnie.py:
```diff
@@ -50,21 +56,25 @@ def setUp(self):
         cmd = ('%s/bonnie++ %s' % (self.srcdir, " ".join(args)))
-        process.run(cmd)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
```
Better debug message can be used.

balamuruhans commented on perf/bonnie.py.data/bonnie.yaml:
```diff
@@ -8,6 +8,7 @@
 
 # Valid options in avocado test are bellow:
```
%s/bellow/below

narasimhan-v commented:
Make the commit message more relevant.

Ex:
Fixing bonnie test

Added -s parameter to fix the "no space left on disk"
Changed the bonnie tool version to latest.

PraveenPenguin commented on perf/bonnie.py.data/Readme.txt:
```diff
@@ -0,0 +1,11 @@
+uid-to-use: root               (user name or it UUID, here it is name i,e root)
+number-to-stat: 10:0:0:2:8192  (Number of files to create file test)
+size_to_pass: 0                (dataset size, here we are passing 0 to skip it as we running file related test)
+
```
some white spaces in README please remove this

PraveenPenguin commented on perf/bonnie.py:
```diff
@@ -36,12 +40,14 @@ class Bonnie(Test):
         """
-        tarball = self.fetch_asset('http://www.coker.com.au/bonnie++/experimental/bonnie++-1.96.tgz')
+        tarball = self.fetch_asset('http://www.coker.com.au/\
+                                    bonnie++/bonnie++-1.03e.tgz')
```
please use expire param in` fetch_asset` as it help to download again it if expire time got expired .

PraveenPenguin commented on perf/bonnie.py:
```diff
@@ -50,21 +56,25 @@ def setUp(self):
         scratch_dir = self.params.get('scratch-dir', default=self.srcdir)
         uid_to_use = self.params.get('uid-to-use', default=None)
         number_to_stat = self.params.get('number-to-stat', default=2048)
+        data_size_to_pass = self.params.get('data_size_to_pass', default=0)
```
as you are extending this can all these definition move to Setup fn ?

PraveenPenguin commented:
@Naresh-ibm i think still review comment not addressed in rebase can you please have a look , also please attach job log

Naresh-ibm commented:
[edited-pass-bonnie.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/664105/edited-pass-bonnie.txt)

Changed the respective files. please review and let me know further if needed anything.
added the log file after edition.

narasimhan-v commented:
Looks good to me

PraveenPenguin commented on perf/bonnie.py:
```diff
@@ -36,12 +38,20 @@ class Bonnie(Test):
+        self.number_to_stat = self.params.get('number-to-stat', default=2048)
+        self.data_size = self.params.get('data_size_to_pass', default=0)
+
+        tarball = self.fetch_asset('http://www.coker.com.au/bonnie++/'
```
please use expire param in fetch_asset

Naresh-ibm replied:
done. and did a git push with changes.

PraveenPenguin commented on perf/bonnie.py.data/Readme.txt:
```diff
@@ -0,0 +1,8 @@
+uid-to-use: root (user name or it UUID, here it is name i,e root)
+number-to-stat: 10:0:0:2:8192 (Number of files to create file test)
+size_to_pass: 0 (dataset size, here we are passing 0 to skip it as we running file related test)
+
```
please remove this white space

Naresh-ibm replied:
done. removed the the white space i,e an empty line.

balamuruhans commented on perf/bonnie.py:
```diff
@@ -50,21 +60,20 @@ def setUp(self):
+        args.append('-d %s' % self.scratch_dir)
+        args.append('-n %s' % self.number_to_stat)
+        args.append('-s %s' % self.data_size)
+        if self.uid_to_use is not None:
```
`if self.uid_to_use:` suffice the condition

Naresh-ibm replied:
Thanks Bala, Changed it.

balamuruhans commented:
LGTM

PraveenPenguin commented:
@narasimhan-v @balamuruhans thanks

PraveenPenguin commented:
@Naresh-ibm thanks looks good :)

Naresh-ibm commented:
@narasimhan-v @balamuruhans @PraveenPenguin  Thank you so much for all your quick responses and reviews. Leant many things in my first commit. Thank you all once again :-)

narasimhan-v commented:
@Naresh-ibm welcome to Upstreaming. Looking forward to many more commits from you !
